### PR TITLE
Release/v1.1.4

### PR DESCRIPTION
**Pull request by msterle**
## v1.1.4 - 2020-01-07
* feat: load secrets from env vars only (if valid)
* feat: add dockerfile for local development using serverless offline
* feat: allow S3 client options to be set for endpoint, addressing style and signature version

### CODE REVIEW

File: src/api_handler.js [hunk 1]
```diff
@@ -60,26 +60,51 @@ const doHandler = (handler, event, context, callback) => {
...
-        return uPortMgr.setSecrets(config)
-      }).then(res => {
-        doHandler(handler, event, context, callback)
+  if (!addressMgr.isSecretsSet() || !linkMgr.isSecretsSet() || !uPortMgr.isSecretsSet()) {
+    // Try setting from environment first
+    addressMgr.setSecrets(envConfig)
+    linkMgr.setSecrets(envConfig)
+    uPortMgr.setSecrets(envConfig)
+      .catch(e => {}) // ignore error since we are retrying below if needed
+      .finally((res) => {
```
Comment (oed) [inline src/api_handler.js]: Why is this needed? (catch, finally).

Comment (msterle) [reply src/api_handler.js]: `uPortMgr` will return a rejected promise if `initIPFS` fails, in which case we need to ignore the error, because we handle it anyways in the check below
```
uPortMgr.isSecretsSet()
```

This package doesn't handle initialization in the most obvious way, but this maintains the existing flow.
